tsdb/tsdbutil/buffer_test.go

```diff
@@ -71,11 +71,11 @@ func TestSampleRing(t *testing.T) {
 						break
 					}
 				}
-				if sold.t >= s.t-c.delta && !found {
-					t.Fatalf("%d: expected sample %d to be in buffer but was not; buffer %v", i, sold.t, buffered)
-				}
-				if sold.t < s.t-c.delta && found {
-					t.Fatalf("%d: unexpected sample %d in buffer; buffer %v", i, sold.t, buffered)
+
+				if sold.t >= s.t-c.delta {
```

Comment:
Would it not be simpler if you had `if found { ... } else { ... }` and the other condition inside assert?